lib/OpenQA/WebAPI/Controller/API/V1/Job.pm

```diff
@@ -253,6 +253,52 @@ sub update_status {
     $self->render(json => $ret);
 }
 
+sub update {
+    my ($self) = @_;
+
+    my $resultset = $self->app->schema->resultset('Jobs');
+    my $job       = $resultset->find(int($self->stash('jobid')))
+      or return $self->render(json => {error => 'Job does not exist'}, status => 404);
+    my $json = $self->req->json
+      or return $self->render(json => {error => 'No updates provided (must be provided as JSON)'}, status => 400);
+    my $settings = delete $json->{settings};
+
+    # validate specified columns
+    my %allowed_cols = map { $_ => 1 } qw(group_id priority retry_avbl);
+    while (my ($key, $value) = each %$json) {
+        return $self->render(json => {error => 'Column ' . $key . ' can not be set'}, status => 400)
+          unless $allowed_cols{$key};
```

Comment:
How would this allow me to check the error case?
* if allowed column in JSON => fine
* if allowed column not in JSON => also fine